include/deal.II/base/tensor.h

```diff
@@ -1841,7 +1849,10 @@ template <class Archive>
 inline void
 Tensor<rank_, dim, Number>::serialize(Archive &ar, const unsigned int)
 {
-  ar &values;
+  if constexpr (rank_ > 1)
+    ar &values;
+  else
+    ar &boost::serialization::make_array(&values[0], dim);
```

Comment:
Let me add here that we could have used the first case also for the `std::array` in the base case (rank==1) but that I expose the array explicitly because I would like to replace the `std::array` in the base case by a `VectorizedArray` and I don't want to change the serialization output again at that point.

(`VectorizedArray` does not have the infrastructure to serialize itself, but in any case, I can only use it when available and need to fall back to `std::array` otherwise. This would lead to platform-dependent serialization output, which I can avoid by just describing the elements here as an explicit array.)

Reply:
I am looking forward to that - albeit I would note that we need to look out for some places, e.g. for what happens with `Tensor<1, dim, VectorizedArray<double>>`. (It should just work fine, because `VectorizedArray<VectorizedArray<double>>` is valid; nonetheless, we might need to have a closer look there.)

Reply:
Right. `std::conditional` is what is necessary, and I know how to do it :-)